api/channel.go

```diff
@@ -234,18 +261,25 @@ func (c *Client) EditChannelPermission(
 
 	return c.FastRequest(
 		"PUT", EndpointChannels+channelID.String()+"/permissions/"+overwriteID.String(),
-		httputil.WithJSONBody(data),
+		httputil.WithJSONBody(data), httputil.WithHeaders(data.Header()),
 	)
 }
 
+type DeleteChannelPermissionData struct {
+	AuditLogReason
+}
+
 // DeleteChannelPermission deletes a channel permission overwrite for a user or
 // role in a channel. Only usable for guild channels.
 //
 // Requires the MANAGE_ROLES permission.
-func (c *Client) DeleteChannelPermission(channelID discord.ChannelID, overwriteID discord.Snowflake) error {
+func (c *Client) DeleteChannelPermission(
+	channelID discord.ChannelID,
+	overwriteID discord.Snowflake, data DeleteChannelPermissionData) error {
```

Comment:
I think just for this case, `data` can just be `reason AuditLogReason` right
away, unless `DeleteChannelPermission` will have more fields in the future.